pycloudlib/ec2/instance.py

```diff
@@ -16,12 +16,12 @@ class EC2Instance(BaseInstance):
 
     _type = 'ec2'
 
-    def __init__(self, client, key_pair, instance):
+    def __init__(self, key_pair, client, instance):
```

Comment:
Given that you are switch order of positional params, I think we need to fix ec2/cloud,py.launch() to properly init Instance now. Also it would be good during that 'fix' to provide named parameters as kwargs for anything we touch so that future refactors don't get bit by positional param re-ordering.

Reply:
If it's part of the base API, I'm actually with not providing default args. The only reason I'm doing this repositioning of args now is because the different APIs weren't consistent with the base. Once everything is consistent there shouldn't be a need for anymore re-orders, and if new parameters are necessary, those can be introduced as kwargs. But I see no need to convert existing args to kwargs.

Reply:
That makes sense thanks.